span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanPreProcessor.java

```diff
@@ -112,6 +131,24 @@ PreProcessedSpan preProcessSpan(Span span) {
       return null;
     }
 
+    // drop the span if the arrival time of it too old than configured threshold
+    long spanProcessedTime = System.currentTimeMillis();
+    long spanStartTime = Timestamps.toMillis(span.getStartTime());
+    Duration spanArrivalDelay =
+        Duration.of(Math.abs(spanProcessedTime - spanStartTime), ChronoUnit.MILLIS);
+
+    if (spanStartTime > 0 && spanArrivalDelay.compareTo(lateArrivalThresholdDuration) > 0) {
```

Comment:
This condition `spanStartTime > 0` is required as default proto value for the field is `0`. So, if there are spans when it is not set value, they are considered valid for backward compatibility.